sycl/include/CL/sycl/detail/device_info.hpp

```diff
@@ -289,6 +290,31 @@ struct get_device_info<vector_class<size_t>,
   }
 };
 
+// Specialization for kernel to kernel pipes.
+// Here we step away from OpenCL, since there is no appropriate cl_device_info
+// enum for global pipes feature. Instead of it we claim, that all and only
+// Intel FPGA devices support kernel to kernel pipe feature (at least at the
+// scope of SYCL_INTEL_data_flow_pipes extension).
+template <>
+struct get_device_info<bool, info::device::kernel_kernel_pipe_support> {
+  static bool _(RT::PiDevice dev) {
+    pi_uint64 device_type;
+    PI_CALL(RT::piDeviceGetInfo(
+        dev, pi::cast<RT::PiDeviceInfo>(info::device::device_type),
+        sizeof(device_type), &device_type, NULL));
+    if (device_type != PI_DEVICE_TYPE_ACC)
+      return false;
+
+    char vendor_name[1024];
+    PI_CALL(RT::piDeviceGetInfo(
+        dev, pi::cast<RT::PiDeviceInfo>(info::device::vendor),
+        sizeof(vendor_name), &vendor_name, NULL));
+
+    return (std::string(vendor_name) == std::string("Intel(R) Corporation"))
```

Comment:
These 3 lines look very tautological... What about just:
```suggestion
    return std::string(vendor_name) == "Intel(R) Corporation";
```

Reply:
Indeed, fixed